src/services/currentUser.js

```diff
@@ -114,40 +112,16 @@ export async function currentUserId(req, res) {
  *
  * This method retrieves the current user details from HSES and finds or creates the TTA Hub user
  */
-export async function retrieveUserDetails(accessToken) {
-  const requestObj = accessToken.sign({
-    method: 'get',
-    url: `${process.env.AUTH_BASE}/auth/user/me`,
-  });
-
-  const { url } = requestObj;
-  const { data } = await axios.get(url, requestObj);
-
+export async function retrieveUserDetails(data) {
   logger.debug(`User details response data: ${JSON.stringify(data, null, 2)}`);
 
-  let name; let username; let userId; let authorities;
-  if (data.principal.attributes) { // PIV card use response
-    name = data.name;
-    username = data.principal.attributes.user.username;
-    userId = data.principal.attributes.user.userId;
-    authorities = data.principal.attributes.user.authorities;
-  } else {
-    name = data.name;
-    username = data.principal.username;
-    userId = data.principal.userId;
-    authorities = data.principal.authorities;
-  }
-
-  let email = null;
-  if (isEmail(username)) {
-    email = username;
-  }
+  const name = [data?.given_name, data?.family_name].filter(Boolean).join(' ') || '';
 
   return findOrCreateUser({
     name,
-    email,
-    hsesUsername: username,
-    hsesAuthorities: authorities.map(({ authority }) => authority),
-    hsesUserId: userId.toString(),
+    email: data?.email?.toString() || '',
+    hsesUsername: data?.sub?.toString() || '',
+    hsesAuthorities: data?.roles || [],
+    hsesUserId: data?.userId?.toString() || '',
```

Comment:
would this create a blank user? or are there DB constraints that prevent that (since '')